modules/pricefeed/handle_periodic_operations.go

```diff
@@ -54,5 +73,27 @@ func (m *Module) updatePrice() error {
 		return fmt.Errorf("error while saving token prices: %s", err)
 	}
 
-	return m.historyModule.UpdatePricesHistory(prices)
+	return nil
+
+}
+
+// updatePricesHistory fetches total amount of coins in the system from RPC
+// and stores historical perice data inside the database
+func (m *Module) updatePricesHistory() error {
+	log.Debug().
+		Str("module", "pricefeed").
+		Str("operation", "pricefeed").
+		Msg("updating token price and market cap history")
+
+	prices, err := m.getTokenPrices()
```

Comment:
I removed the part that forced all the prices to have `LastUpdated` set to `time.Now()` as I think it wasn't correct. @MonikaCat Was there a specific reason that one was there?

Reply:
@RiccardoM `Last Updated` timestamp is fetched from the coingecko and it reflects the time when the price was last updated. We need the exact value at exact time every 1 hour to plot the graph. If price hasn't change within that 1 hour, the returned timestamp will be the same as one hour ago and it will not be stored in db as it will be a duplicated value. So that's why we should pass the current timestamp rather than the timestamp fetched from coingecko.

Reply:
Ok thanks, added back with a comment